transcom/mymove
# MAIN B-21579 Prevent duplicate MS service item

## [B-21579](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21579)

INT PR: https://github.com/transcom/mymove/pull/14086

## Summary

This prevents MS service item from being created again if there is already a MS service item(move management fee.) existing for the move.
Have not found a way to make this happen using the UI, but by using API call I was able to add multiple MS service item.


### How to test

1. Create a HHG move and go through the steps to submit the move to the Prime(Approve and send as TOO).send`.
2. Go to move details page and you should see `Move management' under service items.
3. Use http://officelocal:3000/swagger-ui/ghc.html#/moveTaskOrder/updateMoveTaskOrderStatus swagger to call this api using `ID` and `eTag`(if-match) from dev tools (http://officelocal:3000/ghc/v1/move/3TFWT9) GET request.
Put body: 
```
{
  "serviceCodeCS": false,
  "serviceCodeMS": true
}
```
6. Verify you get a conflict error as response and cannot create the MS service item.


## Screenshots
![image](https://github.com/user-attachments/assets/b3fc53e3-7e4a-4cc4-a3f6-638d672a0ccd)
![image](https://github.com/user-attachments/assets/1a5b8298-6642-4b49-b18d-4fb9d3871eb1)

![image](https://github.com/user-attachments/assets/979e0c38-bf52-4fa1-9cf5-26fafe78b6c6)
